src/key_value_store/client/kvstore_client.cpp

```diff
@@ -50,6 +51,30 @@ std::vector<std::string> KeyValueStoreClient::Get(const std::string &key) {
   return values;
 }
 
+bool KeyValueStoreClient::Get(const std::string &tag,
+                              std::function<bool(std::string)> &callback) {
+  GetRequest request;
+  ClientContext context;
+  context.AddMetadata("request-type", "stream");
+  request.set_key(tag);  // pass tag as key
+
+  // Initializing the ClientReaderWriter
+  std::shared_ptr<ClientReaderWriter<GetRequest, GetReply>> stream(
+      stub_->get(&context));
+
+  // Writing the request to the ReaderWrtier
+  stream->Write(request);
+  stream->WritesDone();
+
+  GetReply response;
+  while (stream->Read(&response)) {
+    callback(response.value());  // might need to capture return value later
+  }
```

Comment:
I'm not 100% sure what this callback is used for. Can you elaborate?

Reply:
This callback will be passed in by the Faz Server layer via caw function.  I am not 100% sure how this will be handled/implemented, so I left that comment of needing to capture the return value.

Reply:
Hmm interesting. I'm not sure why you would need this, because your callback function does the streaming. So on the Faz side, you're storing the callback function and executing that whenever a Put request is performed. 

In any case, I'm approving this PR right now and we can deal with this as you proceed writing your code. 

Also, just a heads up; It seems like you're almost done with your implementation, to meet the project requirements you still need to implement a new protobuf message.